File: backend/app/models/meal.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, Boolean, JSON, UniqueConstraint
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.database import Base

class Meal(Base):
    __tablename__ = "meals"
    __table_args__ = (
        UniqueConstraint("user_id", "client_id", name="uq_meal_user_client_id"),
    )

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    client_id = Column(String, nullable=True, index=True)
    
    name = Column(String, nullable=False)
    meal_quantity = Column(String, nullable=True)
    meal_type = Column(String, nullable=True)
    calories = Column(Integer, default=0)
    protein = Column(Float, default=0.0)
    carbs = Column(Float, default=0.0)
    fat = Column(Float, default=0.0)
    fiber = Column(Float, default=0.0)
    
    timestamp = Column(DateTime(timezone=True), server_default=func.now())
    health_score = Column(JSON, nullable=True)
    is_favorite = Column(Boolean, default=False)
    image_url = Column(String, nullable=True)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    user = relationship("User", back_populates="meals")

# Add back_populates to User model in a separate step or here if possible
# Since User is in another file, we usually do it there.

File: backend/internal/models/meal.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Meal struct {
	ID                uint           `gorm:"primaryKey" json:"id"`
	UserID            string         `gorm:"index" json:"user_id"`
	MealName          string         `json:"meal_name"`
	MealQuantity      string         `json:"meal_quantity"`
	MealType          string         `json:"meal_type"` // breakfast, lunch, dinner, snack
	Calories          int            `json:"calories"`
	Protein           float64        `json:"protein"`
	Carbs             float64        `json:"carbs"`
	Fat               float64        `json:"fat"`
	Fiber             float64        `json:"fiber"`
	Timestamp         time.Time      `json:"timestamp"`
	CalorieConfidence int            `json:"calorie_confidence"`
	Tip               string         `json:"tip"`
	HealthScore       string         `json:"health_score"`
	HealthReason      string         `json:"health_reason"`
	CreatedAt         time.Time      `json:"created_at"`
	UpdatedAt         time.Time      `json:"updated_at"`
	DeletedAt         gorm.DeletedAt `gorm:"index" json:"-"`
}
